pimd/pim6_cmd.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * PIM for IPv6 FRR
+ * Copyright (C) 2022  Vmware, Inc.
+ *		       Mobashshera Rasool <mrasool@vmware.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <zebra.h>
+
+#include "lib/json.h"
+#include "command.h"
+#include "if.h"
+#include "prefix.h"
+#include "zclient.h"
+#include "plist.h"
+#include "hash.h"
+#include "nexthop.h"
+#include "vrf.h"
+#include "ferr.h"
+
+#include "pimd.h"
+#include "pim6_cmd.h"
+#include "pim_vty.h"
+#include "lib/northbound_cli.h"
+#include "pim_errors.h"
+#include "pim_nb.h"
+
+#ifndef VTYSH_EXTRACT_PL
+#include "pimd/pim6_cmd_clippy.c"
+#endif
+
+DEFPY (mld_group_watermark, mld_group_watermark_cmd,
+       "ipv6 mld watermark-warn (1-65535)$limit",
+       IPV6_STR
+       MLD_STR
+       "Configure group limit for watermark warning\n"
+       "Group count to generate watermark warning\n")
+{
+	PIM_DECLVAR_CONTEXT(vrf, pim);
+	/* TBD Depends on MLD data structure changes */
+	return CMD_SUCCESS;
+}
+
+DEFPY (no_mld_group_watermark, no_mld_group_watermark_cmd,
+       "no ipv6 mld watermark-warn [(1-65535)$limit]",
+       NO_STR
+       IPV6_STR
+       MLD_STR
+       "Unconfigure group limit for watermark warning\n"
+       IGNORED_IN_NO_STR)
+{
+	PIM_DECLVAR_CONTEXT(vrf, pim);
+	/* TBD Depends on MLD data structure changes */
+	return CMD_SUCCESS;
+}
+
+DEFUN (interface_ipv6_mld_query_max_response_time,
+       interface_ipv6_mld_query_max_response_time_cmd,
+       "ipv6 mld query-max-response-time (1-65535)",
+       IPV6_STR
+       IFACE_MLD_STR
+       IFACE_MLD_QUERY_MAX_RESPONSE_TIME_STR
+       "Query response value in deci-seconds\n")
+{
+	const struct lyd_node *pim_enable_dnode;
```

Comment:
This code looks almost same to similar IGMP cmd (interface_ip_igmp_query_max_response_time_cmd) except the address-family (frr-routing:ipv6). This comment is common for other MLD CLI cmds as well. 

I think, we should introduce a common file between V4 & V6 and move these code to that file and we can use "#if PIM_IPV == 4" to encode the address-family properly in that common file. That common abstracted function can be called in V4 (pim_cmd.c) & V6 (pim6_cmd.c). 

That will help us maintain common code for both V4 & V6 and any common fixes will be go together. This comment is something similar to @donaldsharp 's suggestion on Socket-changes.

Reply:
Added a detailed comment with e.g. in other PIM PR.
Here is the link : https://github.com/FRRouting/frr/pull/10387/files#r801060701

Reply:
Fixed, Thankyou.